content/en/blog/_posts/2021-12-07-dual-stack-networking-ga.md

```diff
@@ -0,0 +1,47 @@
+---
+layout: blog
+title: 'Kubernetes 1.23: Dual-stack IPv4/IPv6 Networking Reaches GA'
+date: 2021-12-07
+slug: dual-stack-networking-ga
+---
+
+**Author:** Bridget Kromhout (Microsoft)
+
+"When will Kubernetes have IPv6?" This question has been asked with increasing frequency ever since alpha support for IPv6 was first added in K8s 1.9. And while Kubernetes has had IPv6-only clusters since 1.18, migration from IPv4 to IPv6 was not yet possible at that point. At long last, [dual-stack IPv4/IPv6 networking](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/563-dual-stack) reaches general availability (GA) in K8s 1.23. What does dual-stack networking mean for you? Let’s take a look...
+
+
+## Service API updates
+
+Services were single-stack before 1.20, so using both IP families meant creating one Service per IP family. The user experience was simplified in 1.20, when Services were re-implemented to allow both IP families, meaning a single Service can handle both IPv4 and IPv6 workloads. Dual-stack load balancing is possible between services running any combination of IPv4 and IPv6.
+
+The Service API now has new fields to support dual-stack, replacing the single ipFamily field.
+* You can select your choice of IP family by setting `ipFamilyPolicy` to one of three options: SingleStack, PreferDualStack, or RequireDualStack. A service can be changed between single-stack and dual-stack (within some limits).
+* Setting `ipFamilies` to a list of families assigned allows you to set the order of families used.
```

Comment:
I can't fully understand what this sentence mean, can you ellaborate?

Reply:
I'm trying to say that you can choose which IP family to use and, if using both, you can assign the order of families used, by putting them in the desired order. Should I just say that? Or am I missing a nuance?